arrows/vxl/image_io.cxx

```diff
@@ -292,6 +398,10 @@ ::load_(const std::string& filename) const
     {                                                                  \
       typedef vil_pixel_format_type_of<T >::component_type pix_t;      \
       vil_image_view<pix_t> img_pix_t = img_rsc->get_view();           \
+      if( d_->split_channels )                                         \
```

Comment:
I realize that you didn't actually write these macros and the comments that follow are not in any way a suggestion that you make changes but still .. these make me very nervous. Aside from the complete inability to debug them should the need arise, they include several language constructs that I've been warned about including in macros; if statements, multiple lines of execution and all encased in the case statement. They have been around since 2015 so I suspect they are free from the common pitfalls but still, making me nervous.

Considering their age, I wonder if there is some modern c++ way to do the same thing without the need for expanding the code out explicitly. I will investigate that some see if there is a good alternate solution that I can raise in the kwiver working group meeting.

Reply:
What I usually recommend is to make a templated helper function that does the work of macro.  Then the macro doesn't need to be more than a couple line to call that helper with the correct type

Reply:
I addressed this as Matt L. suggested using a templated function in the implementation class.